GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -630,6 +624,17 @@ protected override void OnLoad(EventArgs e)
 
         protected override void OnActivated(EventArgs e)
         {
+            // wait for windows to really be displayed, which isn't necessarily the case in OnLoad()
+            if (_windowsJumpListManager.NeedsJumpListCreation)
```

Comment:
This check does not need to be done here, too, because it is the first thing what `CreateJumpList` does.
So it can be avoided to expose `NeedsJumpListCreation`.

Reply:
Indeed, this wasn't in my first version. However, the call creates three `WindowsThumbnailToolbarButton`s, and I wasn't sure it wouldn't need a handle or other somewhat "expensive" resource (the allocation itself is surely not an issue).